### ISSUE TITLE
Taffybar GTK Window Stops Refreshing (hangs) in WSHUD

### ISSUE THREAD

**Issue by teleshoes** [taffybar/taffybar#228]
the ui sometimes becomes unresponsive when i use wshud instead of wss. it happens seemingly at random, and usually after a full day of uptime, but it happens regularly enough that im certain it only happens in wshud.

its likely a deadlock in the ui or some other concurrency related problem preventing re-painting. clicking things still works, its just that labels and images never change (all widgets, the wshud, the clock, cpu graphs, etc)

i would blame my taffybar, but if i change just one line to switch to WSS, the problem goes away. (i was using WSS for months and then i switched to WSHUD a coupla weeks ago and it started dying regularly again).

i havent (re-?)reported ths bug because i havent been on the LATEST wshud code, but i am now, and it happened twice in the last 3 days. im still trying to collect more data, but i wanted to put this out here now jic

**Comment by teleshoes** [comment]
@IvanMalison 
one more thing; i used to think that this was caused by the slow-image-loading problem, but thats definitely fixed by the latest code.

(changing images quickly doesnt freeze the ui at all, and the image updates as fast as the windows switch. before the latest image-loading commit, the images loaded fairly fast, but they still blocked the ui updating for a tiny bit. further back, before your fix last year, it was really slow AND blocked the ui)

**Comment by IvanMalison** [comment]
@teleshoes Can you give details about which version of ghc you are using? What about your X11 version? What distro are you running? We're getting to the point here with this that I think we have to start looking to these things as potential causes of your issues. Can you build with stack so all of this is 100% reproducible? Does RSS usage go up? Can you post your config so I can run it for a couple of days to see if I can reproduce?

BTW: I have a really hard time believing that you are able to run WSS for multiple days at a time since 
#188 #105 are not fixed in WSS. When I run WSS on my machine I am lucky to get more than an hour of uptime without one of those two causing a crash.

> that labels and images never change

So you are saying it is 100% unresponsive? Or it is it just slow? I can't quite tell from this report.

> slow-image-loading problem,

What is this referring to? I'm not aware of having fixed that. Images are still really slow because they are loaded into lists and not arrays.

> still blocked the ui

Based on this it seems like you are reffing to the rate limiting changes that I added to prevent image updates from hogging cpu?

**Comment by teleshoes** [comment]
> Can you give details about which version..

ghc 8.0.2
xorg 7.7+19
xserver 1.19.3-2
debian sid
linux 4.12.8
xmonad 0.13

> I think we have to start looking to these things as potential causes of your issues

hmm, if i were you, i would expect the problem not to be in display server libs, but in my taffybar config, which is complicated and large.
i would assume that, myself, EXCEPT that the bug only occurs in WSHUD and never in WSS, so at least part of the cause of the bug must be in WSHUD.

> Can you post your config so I can run it for a couple of days to see if I can reproduce?

its huge, and it requires pr208 and pr209, but if you wanna give it a go:
https://github.com/teleshoes/wolke-home-config/tree/master/.config/taffybar

> Do you run taffybar as a systemd service?

no i start it in ~/.xinitrc

> I have a really hard time believing that

i understand your disbelief, but please trust me on this; i routinely run WSS for literal weeks without a crash, and rarely get through two full days with WSHUD.
ive never seen the error from 188, and ive tried hard to get 105 to happen and it just never seems to.

> So you are saying it is 100% unresponsive? Or it is it just slow?

heh, im not quite saying either of those things. it freezes, and it NEVER unfreezes, but it continues to respond to mouse clicks. to be clear, it never changes visually and i have to kill it.

> Images are still really slow

truth, but WSHUD switches icons about as fast as WSS now, and it used to be painfully slow.

> you are reffing to the rate limiting changes

thats the one, yea. that commit made it so that other taffybar widgets continue to update even when wshud is busy loading images. the OTHER fix i mentioned was a bug you fixed from before wshud was merged to master

**Comment by teleshoes** [comment]
```
*** Error in `/home/wolke/.cache/taffybar/taffybar-linux-x86_64': double free or corruption (out): 0x00007f8398006590 ***
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x70bcb)[0x7f83b3e9fbcb]
/lib/x86_64-linux-gnu/libc.so.6(+0x76f96)[0x7f83b3ea5f96]
/lib/x86_64-linux-gnu/libc.so.6(+0x777de)[0x7f83b3ea67de]
/usr/lib/x86_64-linux-gnu/libgdk-x11-2.0.so.0(gdk_region_intersect+0x72)[0x7f83b834cb42]
/usr/lib/x86_64-linux-gnu/libgdk-x11-2.0.so.0(+0x3e347)[0x7f83b8358347]
/usr/lib/x86_64-linux-gnu/libgdk-x11-2.0.so.0(gdk_window_process_all_updates+0x118)[0x7f83b8358ed8]
/usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0(+0xb6bf1)[0x7f83b8685bf1]
/usr/lib/x86_64-linux-gnu/libgdk-x11-2.0.so.0(+0x1dd37)[0x7f83b8337d37]
/lib/x86_64-linux-gnu/libglib-2.0.so.0(g_main_context_dispatch+0x155)[0x7f83b6c94dd5]
/lib/x86_64-linux-gnu/libglib-2.0.so.0(+0x4b1a0)[0x7f83b6c951a0]
/lib/x86_64-linux-gnu/libglib-2.0.so.0(g_main_loop_run+0xc2)[0x7f83b6c954b2]
/usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0(gtk_main+0xb7)[0x7f83b86fe3b7]
/home/wolke/.cache/taffybar/taffybar-linux-x86_64[0x47cde4]
======= Memory map: ========
00400000-01066000 r-xp 00000000 00:2d 46835071                           /home/wolke/.cache/taffybar/taffybar-linux-x86_64
01265000-012be000 r--p 00c65000 00:2d 46835071                           /home/wolke/.cache/taffybar/taffybar-linux-x86_64
012be000-013b7000 rw-p 00cbe000 00:2d 46835071                           /home/wolke/.cache/taffybar/taffybar-linux-x86_64
013b7000-013b9000 rw-p 00000000 00:00 0 
02a43000-02c4e000 rw-p 00000000 00:00 0                                  [heap]
4200000000-4200700000 rw-p 00000000 00:00 0 
4200700000-14200100000 ---p 00000000 00:00 0 
7f8380000000-7f8380021000 rw-p 00000000 00:00 0 
7f8380021000-7f8384000000 ---p 00000000 00:00 0 
7f8384000000-7f8384021000 rw-p 00000000 00:00 0 
7f8384021000-7f8388000000 ---p 00000000 00:00 0 
7f838b7ff000-7f838b800000 ---p 00000000 00:00 0 
7f838b800000-7f838c000000 rw-p 00000000 00:00 0 
7f838c000000-7f838c021000 rw-p 00000000 00:00 0 
7f838c021000-7f8390000000 ---p 00000000 00:00 0 
7f8390000000-7f8390021000 rw-p 00000000 00:00 0 
7f8390021000-7f8394000000 ---p 00000000 00:00 0 
7f8394000000-7f8394025000 rw-p 00000000 00:00 0 
7f8394025000-7f8398000000 ---p 00000000 00:00 0 
7f8398000000-7f8398021000 rw-p 00000000 00:00 0 
7f8398021000-7f839c000000 ---p 00000000 00:00 0 
7f839c000000-7f839c021000 rw-p 00000000 00:00 0 
7f839c021000-7f83a0000000 ---p 00000000 00:00 0 
7f83a0000000-7f83a0021000 rw-p 00000000 00:00 0 
7f83a0021000-7f83a4000000 ---p 00000000 00:00 0 
7f83a4000000-7f83a4021000 rw-p 00000000 00:00 0 
7f83a4021000-7f83a8000000 ---p 00000000 00:00 0 
7f83a8236000-7f83a8237000 ---p 00000000 00:00 0 
7f83a8237000-7f83a8a37000 rw-p 00000000 00:00 0 
7f83a8a37000-7f83a8a38000 ---p 00000000 00:00 0 
7f83a8a38000-7f83a9238000 rw-p 00000000 00:00 0 
7f83a9238000-7f83a9239000 ---p 00000000 00:00 0 
7f83a9239000-7f83a9a39000 rw-p 00000000 00:00 0 
7f83a9a39000-7f83a9a3a000 ---p 00000000 00:00 0 
7f83a9a3a000-7f83aa23a000 rw-p 00000000 00:00 0 
7f83aa23a000-7f83aa23f000 r-xp 00000000 00:15 10581186                   /usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so
7f83aa23f000-7f83aa43e000 ---p 00005000 00:15 10581186                   /usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so
7f83aa43e000-7f83aa43f000 r--p 00004000 00:15 10581186                   /usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so
7f83aa43f000-7f83aa440000 rw-p 00005000 00:15 10581186                   /usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so
7f83aa440000-7f83aa441000 ---p 00000000 00:00 0 
7f83aa441000-7f83aac41000 rw-p 00000000 00:00 0 
7f83aac41000-7f83aac6b000 r-xp 00000000 00:15 23097                      /usr/lib/x86_64-linux-gnu/gtk-2.0/2.10.0/engines/libclearlooks.so
7f83aac6b000-7f83aae6b000 ---p 0002a000 00:15 23097                      /usr/lib/x86_64-linux-gnu/gtk-2.0/2.10.0/engines/libclearlooks.so
7f83aae6b000-7f83aae6c000 r--p 0002a000 00:15 23097                      /usr/lib/x86_64-linux-gnu/gtk-2.0/2.10.0/engines/libclearlooks.so
7f83aae6c000-7f83aae6d000 rw-p 0002b000 00:15 23097                      /usr/lib/x86_64-linux-gnu/gtk-2.0/2.10.0/engines/libclearlooks.so
7f83aae6d000-7f83aae6e000 ---p 00000000 00:00 0 
7f83aae6e000-7f83ab66e000 rw-p 00000000 00:00 0 
7f83ab66e000-7f83ab66f000 ---p 00000000 00:00 0 
7f83ab66f000-7f83abe6f000 rw-p 00000000 00:00 0 
7f83abe6f000-7f83abe70000 ---p 00000000 00:00 0 
7f83abe70000-7f83ac670000 rw-p 00000000 00:00 0 
7f83ac670000-7f83ade97000 r--p 00000000 00:15 12202001                   /usr/lib/locale/locale-archive
7f83ade97000-7f83ade9b000 r-xp 00000000 00:15 12088077                   /lib/x86_64-linux-gnu/libuuid.so.1.3.0
7f83ade9b000-7f83ae09a000 ---p 00004000 00:15 12088077                   /lib/x86_64-linux-gnu/libuuid.so.1.3.0
7f83ae09a000-7f83ae09b000 r--p 00003000 00:15 12088077                   /lib/x86_64-linux-gnu/libuuid.so.1.3.0
7f83ae09b000-7f83ae09c000 rw-p 00004000 00:15 12088077                   /lib/x86_64-linux-gnu/libuuid.so.1.3.0
7f83ae09c000-7f83ae0af000 r-xp 00000000 00:15 11583392                   /lib/x86_64-linux-gnu/libbsd.so.0.8.6
7f83ae0af000-7f83ae2ae000 ---p 00013000 00:15 11583392                   /lib/x86_64-linux-gnu/libbsd.so.0.8.6
7f83ae2ae000-7f83ae2af000 r--p 00012000 00:15 11583392                   /lib/x86_64-linux-gnu/libbsd.so.0.8.6
7f83ae2af000-7f83ae2b0000 rw-p 00013000 00:15 11583392                   /lib/x86_64-linux-gnu/libbsd.so.0.8.6
7f83ae2b0000-7f83ae2b1000 rw-p 00000000 00:00 0 
7f83ae2b1000-7f83ae2c7000 r-xp 00000000 00:15 12091696                   /lib/x86_64-linux-gnu/libgcc_s.so.1
7f83ae2c7000-7f83ae4c6000 ---p 00016000 00:15 12091696                   /lib/x86_64-linux-gnu/libgcc_s.so.1
7f83ae4c6000-7f83ae4c7000 r--p 00015000 00:15 12091696                   /lib/x86_64-linux-gnu/libgcc_s.so.1
7f83ae4c7000-7f83ae4c8000 rw-p 00016000 00:15 12091696                   /lib/x86_64-linux-gnu/libgcc_s.so.1
7f83ae4c8000-7f83ae638000 r-xp 00000000 00:15 12090916                   /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.24
7f83ae638000-7f83ae838000 ---p 00170000 00:15 12090916                   /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.24
7f83ae838000-7f83ae842000 r--p 00170000 00:15 12090916                   /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.24
7f83ae842000-7f83ae844000 rw-p 0017a000 00:15 12090916                   /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.24
7f83ae844000-7f83ae847000 rw-p 00000000 00:00 0 
7f83ae847000-7f83ae84d000 r-xp 00000000 00:15 12134617                   /usr/lib/x86_64-linux-gnu/libdatrie.so.1.3.3
7f83ae84d000-7f83aea4d000 ---p 00006000 00:15 12134617                   /usr/lib/x86_64-linux-gnu/libdatrie.so.1.3.3
7f83aea4d000-7f83aea4e000 r--p 00006000 00:15 12134617                   /usr/lib/x86_64-linux-gnu/libdatrie.so.1.3.3
7f83aea4e000-7f83aea4f000 rw-p 00007000 00:15 12134617                   /usr/lib/x86_64-linux-gnu/libdatrie.so.1.3.3
7f83aea4f000-7f83aea7a000 r-xp 00000000 00:15 11291867                   /usr/lib/x86_64-linux-gnu/libgraphite2.so.3.0.1
7f83aea7a000-7f83aec79000 ---p 0002b000 00:15 11291867                   /usr/lib/x86_64-linux-gnu/libgraphite2.so.3.0.1
7f83aec79000-7f83aec7b000 r--p 0002a000 00:15 11291867                   /usr/lib/x86_64-linux-gnu/libgraphite2.so.3.0.1
7f83aec7b000-7f83aec7c000 rw-p 0002c000 00:15 11291867                   /usr/lib/x86_64-linux-gnu/libgraphite2.so.3.0.1
7f83aec7c000-7f83aecbc000 r-xp 00000000 00:15 12088127                   /lib/x86_64-linux-gnu/libblkid.so.1.1.0
7f83aecbc000-7f83aeebb000 ---p 00040000 00:15 12088127                   /lib/x86_64-linux-gnu/libblkid.so.1.1.0
7f83aeebb000-7f83aeebf000 r--p 0003f000 00:15 12088127                   /lib/x86_64-linux-gnu/libblkid.so.1.1.0
7f83aeebf000-7f83aeec0000 rw-p 00043000 00:15 12088127                   /lib/x86_64-linux-gnu/libblkid.so.1.1.0
7f83aeec0000-7f83aeec1000 rw-p 00000000 00:00 0 
7f83aeec1000-7f83aeec6000 r-xp 00000000 00:15 10574500                   /usr/lib/x86_64-linux-gnu/libXdmcp.so.6.0.0
7f83aeec6000-7f83af0c5000 ---p 00005000 00:15 10574500                   /usr/lib/x86_64-linux-gnu/libXdmcp.so.6.0.0
7f83af0c5000-7f83af0c6000 r--p 00004000 00:15 10574500                   /usr/lib/x86_64-linux-gnu/libXdmcp.so.6.0.0
7f83af0c6000-7f83af0c7000 rw-p 00005000 00:15 10574500                   /usr/lib/x86_64-linux-gnu/libXdmcp.so.6.0.0
7f83af0c7000-7f83af0c9000 r-xp 00000000 00:15 12153156                   /usr/lib/x86_64-linux-gnu/libXau.so.6.0.0
7f83af0c9000-7f83af2c9000 ---p 00002000 00:15 12153156                   /usr/lib/x86_64-linux-gnu/libXau.so.6.0.0
7f83af2c9000-7f83af2ca000 r--p 00002000 00:15 12153156                   /usr/lib/x86_64-linux-gnu/libXau.so.6.0.0
7f83af2ca000-7f83af2cb000 rw-p 00003000 00:15 12153156                   /usr/lib/x86_64-linux-gnu/libXau.so.6.0.0
7f83af2cb000-7f83af2f0000 r-xp 00000000 00:15 11575549                   /lib/x86_64-linux-gnu/liblzma.so.5.2.2
7f83af2f0000-7f83af4ef000 ---p 00025000 00:15 11575549                   /lib/x86_64-linux-gnu/liblzma.so.5.2.2
7f83af4ef000-7f83af4f0000 r--p 00024000 00:15 11575549                   /lib/x86_64-linux-gnu/liblzma.so.5.2.2
7f83af4f0000-7f83af4f1000 rw-p 00025000 00:15 11575549                   /lib/x86_64-linux-gnu/liblzma.so.5.2.2
7f83af4f1000-7f83b0d6d000 r-xp 00000000 00:15 10996919                   /usr/lib/x86_64-linux-gnu/libicudata.so.57.1
7f83b0d6d000-7f83b0f6c000 ---p 0187c000 00:15 10996919                   /usr/lib/x86_64-linux-gnu/libicudata.so.57.1
7f83b0f6c000-7f83b0f6d000 r--p 0187b000 00:15 10996919                   /usr/lib/x86_64-linux-gnu/libicudata.so.57.1
7f83b0f6d000-7f83b0f6e000 rw-p 0187c000 00:15 10996919                   /usr/lib/x86_64-linux-gnu/libicudata.so.57.1
7f83b0f6e000-7f83b1102000 r-xp 00000000 00:15 10996926                   /usr/lib/x86_64-linux-gnu/libicuuc.so.57.1
7f83b1102000-7f83b1301000 ---p 00194000 00:15 10996926                   /usr/lib/x86_64-linux-gnu/libicuuc.so.57.1
7f83b1301000-7f83b1313000 r--p 00193000 00:15 10996926                   /usr/lib/x86_64-linux-gnu/libicuuc.so.57.1
7f83b1313000-7f83b1314000 rw-p 001a5000 00:15 10996926                   /usr/lib/x86_64-linux-gnu/libicuuc.so.57.1
7f83b1314000-7f83b1316000 rw-p 00000000 00:00 0 
7f83b1316000-7f83b1581000 r-xp 00000000 00:15 10996920                   /usr/lib/x86_64-linux-gnu/libicui18n.so.57.1
7f83b1581000-7f83b1780000 ---p 0026b000 00:15 10996920                   /usr/lib/x86_64-linux-gnu/libicui18n.so.57.1
7f83b1780000-7f83b178d000 r--p 0026a000 00:15 10996920                   /usr/lib/x86_64-linux-gnu/libicui18n.so.57.1
7f83b178d000-7f83b178f000 rw-p 00277000 00:15 10996920                   /usr/lib/x86_64-linux-gnu/libicui18n.so.57.1
7f83b178f000-7f83b1790000 rw-p 00000000 00:00 0 
7f83b1790000-7f83b17b8000 r-xp 00000000 00:15 12095491                   /lib/x86_64-linux-gnu/libexpat.so.1.6.5
7f83b17b8000-7f83b19b8000 ---p 00028000 00:15 12095491                   /lib/x86_64-linux-gnu/libexpat.so.1.6.5
7f83b19b8000-7f83b19ba000 r--p 00028000 00:15 12095491                   /lib/x86_64-linux-gnu/libexpat.so.1.6.5
7f83b19ba000-7f83b19bb000 rw-p 0002a000 00:15 12095491                   /lib/x86_64-linux-gnu/libexpat.so.1.6.5
7f83b19bb000-7f83b19c3000 r-xp 00000000 00:15 12134674                   /usr/lib/x86_64-linux-gnu/libthai.so.0.3.0
7f83b19c3000-7f83b1bc3000 ---p 00008000 00:15 12134674                   /usr/lib/x86_64-linux-gnu/libthai.so.0.3.0
7f83b1bc3000-7f83b1bc4000 r--p 00008000 00:15 12134674                   /usr/lib/x86_64-linux-gnu/libthai.so.0.3.0
7f83b1bc4000-7f83b1bc5000 rw-p 00009000 00:15 12134674                   /usr/lib/x86_64-linux-gnu/libthai.so.0.3.0
7f83b1bc5000-7f83b1c58000 r-xp 00000000 00:15 10291310                   /usr/lib/x86_64-linux-gnu/libharfbuzz.so.0.10400.2
7f83b1c58000-7f83b1e57000 ---p 00093000 00:15 10291310                   /usr/lib/x86_64-linux-gnu/libharfbuzz.so.0.10400.2
7f83b1e57000-7f83b1e59000 r--p 00092000 00:15 10291310                   /usr/lib/x86_64-linux-gnu/libharfbuzz.so.0.10400.2
7f83b1e59000-7f83b1e5a000 rw-p 00094000 00:15 10291310                   /usr/lib/x86_64-linux-gnu/libharfbuzz.so.0.10400.2
7f83b1e5a000-7f83b1ea3000 r-xp 00000000 00:15 12088195                   /lib/x86_64-linux-gnu/libmount.so.1.1.0
7f83b1ea3000-7f83b20a3000 ---p 00049000 00:15 12088195                   /lib/x86_64-linux-gnu/libmount.so.1.1.0
7f83b20a3000-7f83b20a4000 r--p 00049000 00:15 12088195                   /lib/x86_64-linux-gnu/libmount.so.1.1.0
7f83b20a4000-7f83b20a5000 rw-p 0004a000 00:15 12088195                   /lib/x86_64-linux-gnu/libmount.so.1.1.0
7f83b20a5000-7f83b20a7000 rw-p 00000000 00:00 0 
7f83b20a7000-7f83b20bb000 r-xp 00000000 00:15 11275869                   /lib/x86_64-linux-gnu/libresolv-2.24.so
7f83b20bb000-7f83b22ba000 ---p 00014000 00:15 11275869                   /lib/x86_64-linux-gnu/libresolv-2.24.so
7f83b22ba000-7f83b22bb000 r--p 00013000 00:15 11275869                   /lib/x86_64-linux-gnu/libresolv-2.24.so
7f83b22bb000-7f83b22bc000 rw-p 00014000 00:15 11275869                   /lib/x86_64-linux-gnu/libresolv-2.24.so
7f83b22bc000-7f83b22be000 rw-p 00000000 00:00 0 
7f83b22be000-7f83b22e3000 r-xp 00000000 00:15 11433258                   /lib/x86_64-linux-gnu/libselinux.so.1
7f83b22e3000-7f83b24e2000 ---p 00025000 00:15 11433258                   /lib/x86_64-linux-gnu/libselinux.so.1
7f83b24e2000-7f83b24e3000 r--p 00024000 00:15 11433258                   /lib/x86_64-linux-gnu/libselinux.so.1
7f83b24e3000-7f83b24e4000 rw-p 00025000 00:15 11433258                   /lib/x86_64-linux-gnu/libselinux.so.1
7f83b24e4000-7f83b24e6000 rw-p 00000000 00:00 0 
7f83b24e6000-7f83b24f2000 r-xp 00000000 00:15 9302864                    /usr/lib/x86_64-linux-gnu/libxcb-render.so.0.0.0
7f83b24f2000-7f83b26f2000 ---p 0000c000 00:15 9302864                    /usr/lib/x86_64-linux-gnu/libxcb-render.so.0.0.0
7f83b26f2000-7f83b26f3000 r--p 0000c000 00:15 9302864                    /usr/lib/x86_64-linux-gnu/libxcb-render.so.0.0.0
7f83b26f3000-7f83b26f4000 rw-p 0000d000 00:15 9302864                    /usr/lib/x86_64-linux-gnu/libxcb-render.so.0.0.0
7f83b26f4000-7f83b271b000 r-xp 00000000 00:15 9302594                    /usr/lib/x86_64-linux-gnu/libxcb.so.1.1.0
7f83b271b000-7f83b291a000 ---p 00027000 00:15 9302594                    /usr/lib/x86_64-linux-gnu/libxcb.so.1.1.0
7f83b291a000-7f83b291b000 r--p 00026000 00:15 9302594                    /usr/lib/x86_64-linux-gnu/libxcb.so.1.1.0
7f83b291b000-7f83b291c000 rw-p 00027000 00:15 9302594                    /usr/lib/x86_64-linux-gnu/libxcb.so.1.1.0
7f83b291c000-7f83b291e000 r-xp 00000000 00:15 9305428                    /usr/lib/x86_64-linux-gnu/libxcb-shm.so.0.0.0
7f83b291e000-7f83b2b1e000 ---p 00002000 00:15 9305428                    /usr/lib/x86_64-linux-gnu/libxcb-shm.so.0.0.0
7f83b2b1e000-7f83b2b1f000 r--p 00002000 00:15 9305428                    /usr/lib/x86_64-linux-gnu/libxcb-shm.so.0.0.0
7f83b2b1f000-7f83b2b20000 rw-p 00003000 00:15 9305428                    /usr/lib/x86_64-linux-gnu/libxcb-shm.so.0.0.0
7f83b2b20000-7f83b2b52000 r-xp 00000000 00:15 12096892                   /usr/lib/x86_64-linux-gnu/libpng16.so.16.31.0
7f83b2b52000-7f83b2d51000 ---p 00032000 00:15 12096892                   /usr/lib/x86_64-linux-gnu/libpng16.so.16.31.0
7f83b2d51000-7f83b2d52000 r--p 00031000 00:15 12096892                   /usr/lib/x86_64-linux-gnu/libpng16.so.16.31.0
7f83b2d52000-7f83b2d53000 rw-p 00032000 00:15 12096892                   /usr/lib/x86_64-linux-gnu/libpng16.so.16.31.0
7f83b2d53000-7f83b2df1000 r-xp 00000000 00:15 9305373                    /usr/lib/x86_64-linux-gnu/libpixman-1.so.0.34.0
7f83b2df1000-7f83b2ff1000 ---p 0009e000 00:15 9305373                    /usr/lib/x86_64-linux-gnu/libpixman-1.so.0.34.0
7f83b2ff1000-7f83b2ff9000 r--p 0009e000 00:15 9305373                    /usr/lib/x86_64-linux-gnu/libpixman-1.so.0.34.0
7f83b2ff9000-7f83b2ffa000 rw-p 000a6000 00:15 9305373                    /usr/lib/x86_64-linux-gnu/libpixman-1.so.0.34.0
7f83b2ffa000-7f83b3003000 r-xp 00000000 00:15 12139347                   /usr/lib/x86_64-linux-gnu/libXcursor.so.1.0.2
7f83b3003000-7f83b3203000 ---p 00009000 00:15 12139347                   /usr/lib/x86_64-linux-gnu/libXcursor.so.1.0.2
7f83b3203000-7f83b3204000 r--p 00009000 00:15 12139347                   /usr/lib/x86_64-linux-gnu/libXcursor.so.1.0.2
7f83b3204000-7f83b3205000 rw-p 0000a000 00:15 12139347                   /usr/lib/x86_64-linux-gnu/libXcursor.so.1.0.2
7f83b3205000-7f83b3214000 r-xp 00000000 00:15 10289109                   /usr/lib/x86_64-linux-gnu/libXi.so.6.1.0
7f83b3214000-7f83b3413000 ---p 0000f000 00:15 10289109                   /usr/lib/x86_64-linux-gnu/libXi.so.6.1.0
7f83b3413000-7f83b3414000 r--p 0000e000 00:15 10289109                   /usr/lib/x86_64-linux-gnu/libXi.so.6.1.0
7f83b3414000-7f83b3415000 rw-p 0000f000 00:15 10289109                   /usr/lib/x86_64-linux-gnu/libXi.so.6.1.0
7f83b3415000-7f83b341e000 r-xp 00000000 00:15 9707100                    /usr/lib/x86_64-linux-gnu/libXrender.so.1.3.0
7f83b341e000-7f83b361d000 ---p 00009000 00:15 9707100                    /usr/lib/x86_64-linux-gnu/libXrender.so.1.3.0
7f83b361d000-7f83b361e000 r--p 00008000 00:15 9707100                    /usr/lib/x86_64-linux-gnu/libXrender.so.1.3.0
7f83b361e000-7f83b361f000 rw-p 00009000 00:15 9707100                    /usr/lib/x86_64-linux-gnu/libXrender.so.1.3.0
7f83b361f000-7f83b3624000 r-xp 00000000 00:15 9716907                    /usr/lib/x86_64-linux-gnu/libXfixes.so.3.1.0
7f83b3624000-7f83b3823000 ---p 00005000 00:15 9716907                    /usr/lib/x86_64-linux-gnu/libXfixes.so.3.1.0
7f83b3823000-7f83b3824000 r--p 00004000 00:15 9716907                    /usr/lib/x86_64-linux-gnu/libXfixes.so.3.1.0
7f83b3824000-7f83b3825000 rw-p 00005000 00:15 9716907                    /usr/lib/x86_64-linux-gnu/libXfixes.so.3.1.0
7f83b3825000-7f83b3827000 r-xp 00000000 00:15 12138332                   /usr/lib/x86_64-linux-gnu/libXdamage.so.1.1.0
7f83b3827000-7f83b3a26000 ---p 00002000 00:15 12138332                   /usr/lib/x86_64-linux-gnu/libXdamage.so.1.1.0
7f83b3a26000-7f83b3a27000 r--p 00001000 00:15 12138332                   /usr/lib/x86_64-linux-gnu/libXdamage.so.1.1.0
7f83b3a27000-7f83b3a28000 rw-p 00002000 00:15 12138332                   /usr/lib/x86_64-linux-gnu/libXdamage.so.1.1.0
7f83b3a28000-7f83b3a2a000 r-xp 00000000 00:15 10288947                   /usr/lib/x86_64-linux-gnu/libXcomposite.so.1.0.0
7f83b3a2a000-7f83b3c29000 ---p 00002000 00:15 10288947                   /usr/lib/x86_64-linux-gnu/libXcomposite.so.1.0.0
7f83b3c29000-7f83b3c2a000 r--p 00001000 00:15 10288947                   /usr/lib/x86_64-linux-gnu/libXcomposite.so.1.0.0
7f83b3c2a000-7f83b3c2b000 rw-p 00002000 00:15 10288947                   /usr/lib/x86_64-linux-gnu/libXcomposite.so.1.0.0
7f83b3c2b000-7f83b3c2e000 r-xp 00000000 00:15 12099607                   /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0.5306.0
7f83b3c2e000-7f83b3e2d000 ---p 00003000 00:15 12099607                   /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0.5306.0
7f83b3e2d000-7f83b3e2e000 r--p 00002000 00:15 12099607                   /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0.5306.0
7f83b3e2e000-7f83b3e2f000 rw-p 00003000 00:15 12099607                   /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0.5306.0
7f83b3e2f000-7f83b3fc4000 r-xp 00000000 00:15 11275853                   /lib/x86_64-linux-gnu/libc-2.24.so
7f83b3fc4000-7f83b41c4000 ---p 00195000 00:15 11275853                   /lib/x86_64-linux-gnu/libc-2.24.so
7f83b41c4000-7f83b41c8000 r--p 00195000 00:15 11275853                   /lib/x86_64-linux-gnu/libc-2.24.so
7f83b41c8000-7f83b41ca000 rw-p 00199000 00:15 11275853                   /lib/x86_64-linux-gnu/libc-2.24.so
7f83b41ca000-7f83b41ce000 rw-p 00000000 00:00 0 
7f83b41ce000-7f83b41d5000 r-xp 00000000 00:15 9288783                    /usr/lib/x86_64-linux-gnu/libffi.so.6.0.4
7f83b41d5000-7f83b43d5000 ---p 00007000 00:15 9288783                    /usr/lib/x86_64-linux-gnu/libffi.so.6.0.4
7f83b43d5000-7f83b43d6000 r--p 00007000 00:15 9288783                    /usr/lib/x86_64-linux-gnu/libffi.so.6.0.4
7f83b43d6000-7f83b43d7000 rw-p 00008000 00:15 9288783                    /usr/lib/x86_64-linux-gnu/libffi.so.6.0.4
7f83b43d7000-7f83b44da000 r-xp 00000000 00:15 11275857                   /lib/x86_64-linux-gnu/libm-2.24.so
7f83b44da000-7f83b46d9000 ---p 00103000 00:15 11275857                   /lib/x86_64-linux-gnu/libm-2.24.so
7f83b46d9000-7f83b46da000 r--p 00102000 00:15 11275857                   /lib/x86_64-linux-gnu/libm-2.24.so
7f83b46da000-7f83b46db000 rw-p 00103000 00:15 11275857                   /lib/x86_64-linux-gnu/libm-2.24.so
7f83b46db000-7f83b475c000 r-xp 00000000 00:15 9913819                    /usr/lib/x86_64-linux-gnu/libgmp.so.10.3.2
7f83b475c000-7f83b495c000 ---p 00081000 00:15 9913819                    /usr/lib/x86_64-linux-gnu/libgmp.so.10.3.2
7f83b495c000-7f83b495d000 r--p 00081000 00:15 9913819                    /usr/lib/x86_64-linux-gnu/libgmp.so.10.3.2
7f83b495d000-7f83b495e000 rw-p 00082000 00:15 9913819                    /usr/lib/x86_64-linux-gnu/libgmp.so.10.3.2
7f83b495e000-7f83b4977000 r-xp 00000000 00:15 10270534                   /lib/x86_64-linux-gnu/libz.so.1.2.8
7f83b4977000-7f83b4b76000 ---p 00019000 00:15 10270534                   /lib/x86_64-linux-gnu/libz.so.1.2.8
7f83b4b76000-7f83b4b77000 r--p 00018000 00:15 10270534                   /lib/x86_64-linux-gnu/libz.so.1.2.8
7f83b4b77000-7f83b4b78000 rw-p 00019000 00:15 10270534                   /lib/x86_64-linux-gnu/libz.so.1.2.8
7f83b4b78000-7f83b4b90000 r-xp 00000000 00:15 11275868                   /lib/x86_64-linux-gnu/libpthread-2.24.so
7f83b4b90000-7f83b4d8f000 ---p 00018000 00:15 11275868                   /lib/x86_64-linux-gnu/libpthread-2.24.so
7f83b4d8f000-7f83b4d90000 r--p 00017000 00:15 11275868                   /lib/x86_64-linux-gnu/libpthread-2.24.so
7f83b4d90000-7f83b4d91000 rw-p 00018000 00:15 11275868                   /lib/x86_64-linux-gnu/libpthread-2.24.so
7f83b4d91000-7f83b4d95000 rw-p 00000000 00:00 0 
7f83b4d95000-7f83b4d98000 r-xp 00000000 00:15 11275856                   /lib/x86_64-linux-gnu/libdl-2.24.so
7f83b4d98000-7f83b4f97000 ---p 00003000 00:15 11275856                   /lib/x86_64-linux-gnu/libdl-2.24.so
7f83b4f97000-7f83b4f98000 r--p 00002000 00:15 11275856                   /lib/x86_64-linux-gnu/libdl-2.24.so
7f83b4f98000-7f83b4f99000 rw-p 00003000 00:15 11275856                   /lib/x86_64-linux-gnu/libdl-2.24.so
7f83b4f99000-7f83b4f9b000 r-xp 00000000 00:15 11275872                   /lib/x86_64-linux-gnu/libutil-2.24.so
7f83b4f9b000-7f83b519a000 ---p 00002000 00:15 11275872                   /lib/x86_64-linux-gnu/libutil-2.24.so
7f83b519a000-7f83b519b000 r--p 00001000 00:15 11275872                   /lib/x86_64-linux-gnu/libutil-2.24.so
7f83b519b000-7f83b519c000 rw-p 00002000 00:15 11275872                   /lib/x86_64-linux-gnu/libutil-2.24.so
7f83b519c000-7f83b51a3000 r-xp 00000000 00:15 11275870                   /lib/x86_64-linux-gnu/librt-2.24.so
7f83b51a3000-7f83b53a2000 ---p 00007000 00:15 11275870                   /lib/x86_64-linux-gnu/librt-2.24.so
7f83b53a2000-7f83b53a3000 r--p 00006000 00:15 11275870                   /lib/x86_64-linux-gnu/librt-2.24.so
7f83b53a3000-7f83b53a4000 rw-p 00007000 00:15 11275870                   /lib/x86_64-linux-gnu/librt-2.24.so
7f83b53a4000-7f83b53a5000 r-xp 00000000 00:15 12099609                   /usr/lib/x86_64-linux-gnu/libgthread-2.0.so.0.5306.0
7f83b53a5000-7f83b55a4000 ---p 00001000 00:15 12099609                   /usr/lib/x86_64-linux-gnu/libgthread-2.0.so.0.5306.0
7f83b55a4000-7f83b55a5000 r--p 00000000 00:15 12099609                   /usr/lib/x86_64-linux-gnu/libgthread-2.0.so.0.5306.0
7f83b55a5000-7f83b55a6000 rw-p 00001000 00:15 12099609                   /usr/lib/x86_64-linux-gnu/libgthread-2.0.so.0.5306.0
7f83b55a6000-7f83b5618000 r-xp 00000000 00:15 11943892                   /lib/x86_64-linux-gnu/libpcre.so.3.13.3
7f83b5618000-7f83b5817000 ---p 00072000 00:15 11943892                   /lib/x86_64-linux-gnu/libpcre.so.3.13.3
7f83b5817000-7f83b5818000 r--p 00071000 00:15 11943892                   /lib/x86_64-linux-gnu/libpcre.so.3.13.3
7f83b5818000-7f83b5819000 rw-p 00072000 00:15 11943892                   /lib/x86_64-linux-gnu/libpcre.so.3.13.3
7f83b5819000-7f83b5823000 r-xp 00000000 00:15 9741829                    /usr/lib/x86_64-linux-gnu/libXrandr.so.2.2.0
7f83b5823000-7f83b5a22000 ---p 0000a000 00:15 9741829                    /usr/lib/x86_64-linux-gnu/libXrandr.so.2.2.0
7f83b5a22000-7f83b5a23000 r--p 00009000 00:15 9741829                    /usr/lib/x86_64-linux-gnu/libXrandr.so.2.2.0
7f83b5a23000-7f83b5a24000 rw-p 0000a000 00:15 9741829                    /usr/lib/x86_64-linux-gnu/libXrandr.so.2.2.0
7f83b5a24000-7f83b5a35000 r-xp 00000000 00:15 10767870                   /usr/lib/x86_64-linux-gnu/libXext.so.6.4.0
7f83b5a35000-7f83b5c34000 ---p 00011000 00:15 10767870                   /usr/lib/x86_64-linux-gnu/libXext.so.6.4.0
7f83b5c34000-7f83b5c35000 r--p 00010000 00:15 10767870                   /usr/lib/x86_64-linux-gnu/libXext.so.6.4.0
7f83b5c35000-7f83b5c36000 rw-p 00011000 00:15 10767870                   /usr/lib/x86_64-linux-gnu/libXext.so.6.4.0
7f83b5c36000-7f83b5c38000 r-xp 00000000 00:15 10777710                   /usr/lib/x86_64-linux-gnu/libXinerama.so.1.0.0
7f83b5c38000-7f83b5e37000 ---p 00002000 00:15 10777710                   /usr/lib/x86_64-linux-gnu/libXinerama.so.1.0.0
7f83b5e37000-7f83b5e38000 r--p 00001000 00:15 10777710                   /usr/lib/x86_64-linux-gnu/libXinerama.so.1.0.0
7f83b5e38000-7f83b5e39000 rw-p 00002000 00:15 10777710                   /usr/lib/x86_64-linux-gnu/libXinerama.so.1.0.0
7f83b5e39000-7f83b5ff0000 r-xp 00000000 00:15 12097007                   /usr/lib/x86_64-linux-gnu/libxml2.so.2.9.4
7f83b5ff0000-7f83b61f0000 ---p 001b7000 00:15 12097007                   /usr/lib/x86_64-linux-gnu/libxml2.so.2.9.4
7f83b61f0000-7f83b61f8000 r--p 001b7000 00:15 12097007                   /usr/lib/x86_64-linux-gnu/libxml2.so.2.9.4
7f83b61f8000-7f83b61fa000 rw-p 001bf000 00:15 12097007                   /usr/lib/x86_64-linux-gnu/libxml2.so.2.9.4
7f83b61fa000-7f83b61fb000 rw-p 00000000 00:00 0 
7f83b61fb000-7f83b6335000 r-xp 00000000 00:15 10274643                   /usr/lib/x86_64-linux-gnu/libX11.so.6.3.0
7f83b6335000-7f83b6535000 ---p 0013a000 00:15 10274643                   /usr/lib/x86_64-linux-gnu/libX11.so.6.3.0
7f83b6535000-7f83b6536000 r--p 0013a000 00:15 10274643                   /usr/lib/x86_64-linux-gnu/libX11.so.6.3.0
7f83b6536000-7f83b653b000 rw-p 0013b000 00:15 10274643                   /usr/lib/x86_64-linux-gnu/libX11.so.6.3.0
7f83b653b000-7f83b6550000 r-xp 00000000 00:15 10796589                   /usr/lib/x86_64-linux-gnu/libXft.so.2.3.2
7f83b6550000-7f83b674f000 ---p 00015000 00:15 10796589                   /usr/lib/x86_64-linux-gnu/libXft.so.2.3.2
7f83b674f000-7f83b6750000 r--p 00014000 00:15 10796589                   /usr/lib/x86_64-linux-gnu/libXft.so.2.3.2
7f83b6750000-7f83b6751000 rw-p 00015000 00:15 10796589                   /usr/lib/x86_64-linux-gnu/libXft.so.2.3.2
7f83b6751000-7f83b67ff000 r-xp 00000000 00:15 11436615                   /usr/lib/x86_64-linux-gnu/libfreetype.so.6.14.0
7f83b67ff000-7f83b69fe000 ---p 000ae000 00:15 11436615                   /usr/lib/x86_64-linux-gnu/libfreetype.so.6.14.0
7f83b69fe000-7f83b6a05000 r--p 000ad000 00:15 11436615                   /usr/lib/x86_64-linux-gnu/libfreetype.so.6.14.0
7f83b6a05000-7f83b6a06000 rw-p 000b4000 00:15 11436615                   /usr/lib/x86_64-linux-gnu/libfreetype.so.6.14.0
7f83b6a06000-7f83b6a44000 r-xp 00000000 00:15 11437151                   /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.9.3
7f83b6a44000-7f83b6c43000 ---p 0003e000 00:15 11437151                   /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.9.3
7f83b6c43000-7f83b6c45000 r--p 0003d000 00:15 11437151                   /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.9.3
7f83b6c45000-7f83b6c4a000 rw-p 0003f000 00:15 11437151                   /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.9.3
7f83b6c4a000-7f83b6d5b000 r-xp 00000000 00:15 12099602                   /lib/x86_64-linux-gnu/libglib-2.0.so.0.5306.0
7f83b6d5b000-7f83b6f5b000 ---p 00111000 00:15 12099602                   /lib/x86_64-linux-gnu/libglib-2.0.so.0.5306.0
7f83b6f5b000-7f83b6f5c000 r--p 00111000 00:15 12099602                   /lib/x86_64-linux-gnu/libglib-2.0.so.0.5306.0
7f83b6f5c000-7f83b6f5d000 rw-p 00112000 00:15 12099602                   /lib/x86_64-linux-gnu/libglib-2.0.so.0.5306.0
7f83b6f5d000-7f83b6f5e000 rw-p 00000000 00:00 0 
7f83b6f5e000-7f83b6fb0000 r-xp 00000000 00:15 12099608                   /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.5306.0
7f83b6fb0000-7f83b71b0000 ---p 00052000 00:15 12099608                   /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.5306.0
7f83b71b0000-7f83b71b1000 r--p 00052000 00:15 12099608                   /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.5306.0
7f83b71b1000-7f83b71b2000 rw-p 00053000 00:15 12099608                   /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.5306.0
7f83b71b2000-7f83b71fc000 r-xp 00000000 00:15 12134450                   /usr/lib/x86_64-linux-gnu/libpango-1.0.so.0.4000.9
7f83b71fc000-7f83b73fc000 ---p 0004a000 00:15 12134450                   /usr/lib/x86_64-linux-gnu/libpango-1.0.so.0.4000.9
7f83b73fc000-7f83b73ff000 r--p 0004a000 00:15 12134450                   /usr/lib/x86_64-linux-gnu/libpango-1.0.so.0.4000.9
7f83b73ff000-7f83b7400000 rw-p 0004d000 00:15 12134450                   /usr/lib/x86_64-linux-gnu/libpango-1.0.so.0.4000.9
7f83b7400000-7f83b7414000 r-xp 00000000 00:15 12134490                   /usr/lib/x86_64-linux-gnu/libpangoft2-1.0.so.0.4000.9
7f83b7414000-7f83b7614000 ---p 00014000 00:15 12134490                   /usr/lib/x86_64-linux-gnu/libpangoft2-1.0.so.0.4000.9
7f83b7614000-7f83b7615000 r--p 00014000 00:15 12134490                   /usr/lib/x86_64-linux-gnu/libpangoft2-1.0.so.0.4000.9
7f83b7615000-7f83b7616000 rw-p 00015000 00:15 12134490                   /usr/lib/x86_64-linux-gnu/libpangoft2-1.0.so.0.4000.9
7f83b7616000-7f83b77a7000 r-xp 00000000 00:15 12099606                   /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0.5306.0
7f83b77a7000-7f83b79a6000 ---p 00191000 00:15 12099606                   /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0.5306.0
7f83b79a6000-7f83b79ad000 r--p 00190000 00:15 12099606                   /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0.5306.0
7f83b79ad000-7f83b79ae000 rw-p 00197000 00:15 12099606                   /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0.5306.0
7f83b79ae000-7f83b79b0000 rw-p 00000000 00:00 0 
7f83b79b0000-7f83b79d2000 r-xp 00000000 00:15 10581194                   /usr/lib/x86_64-linux-gnu/libgdk_pixbuf-2.0.so.0.3600.5
7f83b79d2000-7f83b7bd1000 ---p 00022000 00:15 10581194                   /usr/lib/x86_64-linux-gnu/libgdk_pixbuf-2.0.so.0.3600.5
7f83b7bd1000-7f83b7bd2000 r--p 00021000 00:15 10581194                   /usr/lib/x86_64-linux-gnu/libgdk_pixbuf-2.0.so.0.3600.5
7f83b7bd2000-7f83b7bd3000 rw-p 00022000 00:15 10581194                   /usr/lib/x86_64-linux-gnu/libgdk_pixbuf-2.0.so.0.3600.5
7f83b7bd3000-7f83b7ce1000 r-xp 00000000 00:15 11441025                   /usr/lib/x86_64-linux-gnu/libcairo.so.2.11400.10
7f83b7ce1000-7f83b7ee1000 ---p 0010e000 00:15 11441025                   /usr/lib/x86_64-linux-gnu/libcairo.so.2.11400.10
7f83b7ee1000-7f83b7ee4000 r--p 0010e000 00:15 11441025                   /usr/lib/x86_64-linux-gnu/libcairo.so.2.11400.10
7f83b7ee4000-7f83b7ee5000 rw-p 00111000 00:15 11441025                   /usr/lib/x86_64-linux-gnu/libcairo.so.2.11400.10
7f83b7ee5000-7f83b7ee7000 rw-p 00000000 00:00 0 
7f83b7ee7000-7f83b7f0a000 r-xp 00000000 00:15 11453358                   /usr/lib/x86_64-linux-gnu/libatk-1.0.so.0.22409.1
7f83b7f0a000-7f83b8109000 ---p 00023000 00:15 11453358                   /usr/lib/x86_64-linux-gnu/libatk-1.0.so.0.22409.1
7f83b8109000-7f83b810c000 r--p 00022000 00:15 11453358                   /usr/lib/x86_64-linux-gnu/libatk-1.0.so.0.22409.1
7f83b810c000-7f83b810d000 rw-p 00025000 00:15 11453358                   /usr/lib/x86_64-linux-gnu/libatk-1.0.so.0.22409.1
7f83b810d000-7f83b8119000 r-xp 00000000 00:15 12134550                   /usr/lib/x86_64-linux-gnu/libpangocairo-1.0.so.0.4000.9
7f83b8119000-7f83b8318000 ---p 0000c000 00:15 12134550                   /usr/lib/x86_64-linux-gnu/libpangocairo-1.0.so.0.4000.9
7f83b8318000-7f83b8319000 r--p 0000b000 00:15 12134550                   /usr/lib/x86_64-linux-gnu/libpangocairo-1.0.so.0.4000.9
7f83b8319000-7f83b831a000 rw-p 0000c000 00:15 12134550                   /usr/lib/x86_64-linux-gnu/libpangocairo-1.0.so.0.4000.9
7f83b831a000-7f83b83ca000 r-xp 00000000 00:15 10291109                   /usr/lib/x86_64-linux-gnu/libgdk-x11-2.0.so.0.2400.31
7f83b83ca000-7f83b85c9000 ---p 000b0000 00:15 10291109                   /usr/lib/x86_64-linux-gnu/libgdk-x11-2.0.so.0.2400.31
7f83b85c9000-7f83b85cd000 r--p 000af000 00:15 10291109                   /usr/lib/x86_64-linux-gnu/libgdk-x11-2.0.so.0.2400.31
7f83b85cd000-7f83b85cf000 rw-p 000b3000 00:15 10291109                   /usr/lib/x86_64-linux-gnu/libgdk-x11-2.0.so.0.2400.31
7f83b85cf000-7f83b8a0b000 r-xp 00000000 00:15 10291110                   /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.31
7f83b8a0b000-7f83b8c0a000 ---p 0043c000 00:15 10291110                   /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.31
7f83b8c0a000-7f83b8c11000 r--p 0043b000 00:15 10291110                   /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.31
7f83b8c11000-7f83b8c15000 rw-p 00442000 00:15 10291110                   /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.31
7f83b8c15000-7f83b8c18000 rw-p 00000000 00:00 0 
7f83b8c18000-7f83b8c3b000 r-xp 00000000 00:15 11275849                   /lib/x86_64-linux-gnu/ld-2.24.so
7f83b8c6b000-7f83b8c6c000 rw-p 00000000 00:00 0 
7f83b8c6c000-7f83b8cc0000 r--p 00000000 00:15 8724990                    /usr/share/fonts/truetype/dejavu/DejaVuSansMono.ttf
7f83b8cc0000-7f83b8d20000 rw-s 00000000 00:05 2949120                    /SYSV00000000 (deleted)
7f83b8d20000-7f83b8d31000 r--p 00000000 00:15 153710                     /usr/share/fonts/truetype/inconsolata/Inconsolata-Bold.ttf
7f83b8d31000-7f83b8d40000 r--p 00000000 00:15 11822802                   /usr/share/fonts/qtbigtext/Inconsolata.otf
7f83b8d40000-7f83b8d47000 r--s 00000000 00:15 11520971                   /var/cache/fontconfig/4be9850f182b35c1350b6bbf2e42601c-le64.cache-7
7f83b8d47000-7f83b8d48000 r--s 00000000 00:15 11520985                   /var/cache/fontconfig/087e1975ba9a574b140bb1df193bf770-le64.cache-7
7f83b8d48000-7f83b8d4c000 r--s 00000000 00:15 12202094                   /var/cache/fontconfig/6aa41aa22e18b8fa06a12da28ea9c28b-le64.cache-7
7f83b8d4c000-7f83b8d57000 r--s 00000000 00:15 11520979                   /var/cache/fontconfig/945677eb7aeaf62f1d50efc3fb3ec7d8-le64.cache-7
7f83b8d57000-7f83b8d5a000 r--s 00000000 00:15 11990950                   /var/cache/fontconfig/f24b2111ab8703b4e963115a8cf14259-le64.cache-7
7f83b8d5a000-7f83b8d5c000 r--s 00000000 00:15 11520973                   /var/cache/fontconfig/5d7ab74b99a79b15fc02249096da7fe0-le64.cache-7
7f83b8d5c000-7f83b8d5d000 r--s 00000000 00:15 11990948                   /var/cache/fontconfig/5d1cca7074f29429a8d18692746c2426-le64.cache-7
7f83b8d5d000-7f83b8d5e000 r--s 00000000 00:15 12202092                   /var/cache/fontconfig/4794a0821666d79190d59a36cb4f44b5-le64.cache-7
7f83b8d5e000-7f83b8d86000 r--s 00000000 00:15 11520965                   /var/cache/fontconfig/0bd3dc0958fa2205aaaa8ebb13e2872b-le64.cache-7
7f83b8d86000-7f83b8d8c000 r--s 00000000 00:15 11520963                   /var/cache/fontconfig/3047814df9a2f067bd2d96a2b9c36e5a-le64.cache-7
7f83b8d8c000-7f83b8d94000 r--s 00000000 00:15 11520961                   /var/cache/fontconfig/bf3b770c553c462765856025a94f1ce6-le64.cache-7
7f83b8d94000-7f83b8d95000 r--s 00000000 00:15 11520959                   /var/cache/fontconfig/89034621ae2a8922916bb6bfa5799546-le64.cache-7
7f83b8d95000-7f83b8d98000 r--s 00000000 00:15 11520957                   /var/cache/fontconfig/e49e89034d371f0f9de17aab02136486-le64.cache-7
7f83b8d98000-7f83b8dab000 r--s 00000000 00:15 11520955                   /var/cache/fontconfig/d52a8644073d54c13679302ca1180695-le64.cache-7
7f83b8dab000-7f83b8dac000 r--s 00000000 00:15 11520953                   /var/cache/fontconfig/8a687c406b77f27d99abfeeba937fcce-le64.cache-7
7f83b8dac000-7f83b8db8000 r--s 00000000 00:15 11520949                   /var/cache/fontconfig/d589a48862398ed80a3d6066f4f56f4c-le64.cache-7
7f83b8db8000-7f83b8dd9000 r--s 00000000 00:15 11520989                   /var/cache/fontconfig/467c019e582ee353435ea5c21d137ef6-le64.cache-7
7f83b8dd9000-7f83b8df6000 rw-p 00000000 00:00 0 
7f83b8df6000-7f83b8df7000 r--s 00000000 00:15 11520951                   /var/cache/fontconfig/3f7329c5293ffd510edef78f73874cfd-le64.cache-7
7f83b8df7000-7f83b8df8000 r--s 00000000 00:15 11990946                   /var/cache/fontconfig/e7de81b01590fb7e12b38e274e17d0db-le64.cache-7
7f83b8df8000-7f83b8e01000 r--s 00000000 00:15 11520941                   /var/cache/fontconfig/d0972c3d32f097851eb916381fc38920-le64.cache-7
7f83b8e01000-7f83b8e08000 r--s 00000000 00:15 11520939                   /var/cache/fontconfig/53d14c92082a93e67d5078324eb314ca-le64.cache-7
7f83b8e08000-7f83b8e09000 r--s 00000000 00:15 11520937                   /var/cache/fontconfig/c277e94e32b20404286a1ddafa5a80f0-le64.cache-7
7f83b8e09000-7f83b8e0c000 r--s 00000000 00:15 11520909                   /var/cache/fontconfig/e13b20fdb08344e0e664864cc2ede53d-le64.cache-7
7f83b8e0c000-7f83b8e0e000 r--s 00000000 00:15 12051219                   /var/cache/fontconfig/7ef2298fde41cc6eeb7af42e48b7d293-le64.cache-7
7f83b8e0e000-7f83b8e0f000 r--s 00000000 00:15 11822821                   /var/cache/fontconfig/63ab64cbc847c652737466bb5bacf0e2-le64.cache-7
7f83b8e0f000-7f83b8e10000 r--s 00000000 00:15 11520935                   /var/cache/fontconfig/573ec803664ed168555e0e8b6d0f0c7f-le64.cache-7
7f83b8e10000-7f83b8e11000 r--s 00000000 00:15 11520991                   /var/cache/fontconfig/e0aa53bcfa504e64f87823c16bc01eb6-le64.cache-7
7f83b8e11000-7f83b8e12000 rwxp 00000000 00:00 0 
7f83b8e12000-7f83b8e36000 r--s 00000000 00:15 12201192                   /usr/share/mime/mime.cache
7f83b8e36000-7f83b8e38000 r--s 00000000 00:2d 28059143                   /home/wolke/.local/share/mime/mime.cache
7f83b8e38000-7f83b8e3b000 rw-p 00000000 00:00 0 
7f83b8e3b000-7f83b8e3c000 r--p 00023000 00:15 11275849                   /lib/x86_64-linux-gnu/ld-2.24.so
7f83b8e3c000-7f83b8e3d000 rw-p 00024000 00:15 11275849                   /lib/x86_64-linux-gnu/ld-2.24.so
7f83b8e3d000-7f83b8e3e000 rw-p 00000000 00:00 0 
7ffe004ff000-7ffe00520000 rw-p 00000000 00:00 0                          [stack]
7ffe00574000-7ffe00577000 r--p 00000000 00:00 0                          [vvar]
7ffe00577000-7ffe00579000 r-xp 00000000 00:00 0                          [vdso]
```

**Comment by teleshoes** [comment]
@IvanMalison 

below is from my taffybar log, not my xmonad log (stdout+stderr from taffybar exec)

this, or the above, SEEMS to precede my crash, and afaict, never happens when i use WSS.


```
xmonad: X11 error: BadDrawable (invalid Pixmap or Window parameter), request code=14, error code=9
taffybar-linux-x86_64: user error (Error in function getGeometry)
```

**Comment by IvanMalison** [comment]
@teleshoes Oh that is a relatively new addition of a feature of WSHUD that wss does not have, so it makes sense that this would not occur in WSS.

You should be able to disable the call to getGeometry by setting the iconSort of the HUD config to

``` haskell
return . id
```

I suspect, however, that this error is simply a manifestation of #105. It says "Bad drawable" because the window about which information is being requested no longer exists.

**Comment by IvanMalison** [comment]
The first error you posted seems to be in inside of gtk AFTER it has dropped in to C. That is obviously pretty weird, especially since it just seems to be in the main gtk loop. I really have no idea how WSHUD would be causing that.

I occasionally get this error, which is somewhat similar, I suppose:

```
Sep 26 15:26:27 imalison-arch env[9787]: Gdk:ERROR:gdkregion-generic.c:1453:miSubtractO: assertion failed: (x1<r1->x2)
Sep 26 15:26:27 imalison-arch systemd[1150]: taffybar.service: Main process exited, code=exited, status=134/n/a
Sep 26 15:26:27 imalison-arch systemd[1150]: taffybar.service: Unit entered failed state.
Sep 26 15:26:27 imalison-arch systemd[1150]: taffybar.service: Failed with result 'exit-code'.
Sep 26 15:26:27 imalison-arch systemd-coredump[14552]: Process 9946 (taffybar-linux-) of user 1000 dumped core.
                                                       
                                                       Stack trace of thread 9946:
                                                       #0  0x00007f678525f8a0 raise (libc.so.6)
                                                       #1  0x00007f6785260f09 abort (libc.so.6)
                                                       #2  0x00007f6787e64cbd g_assertion_message (libglib-2.0.so.0)
                                                       #3  0x00007f6787e64d4a g_assertion_message_expr (libglib-2.0.so.0)
                                                       #4  0x00007f6789510511 n/a (libgdk-x11-2.0.so.0)
                                                       #5  0x00007f6789510d0a n/a (libgdk-x11-2.0.so.0)
                                                       #6  0x00007f6789511820 gdk_region_subtract (libgdk-x11-2.0.so.0)
                                                       #7  0x00007f6789520226 n/a (libgdk-x11-2.0.so.0)
                                                       #8  0x00007f678951cd71 n/a (libgdk-x11-2.0.so.0)
                                                       #9  0x00007f678951d688 gdk_window_process_all_updates (libgdk-x11-2.0.so.0)
                                                       #10 0x00007f678951d6e9 n/a (libgdk-x11-2.0.so.0)
                                                       #11 0x00007f67894fc4b7 n/a (libgdk-x11-2.0.so.0)
                                                       #12 0x00007f6787e3e8c5 g_main_context_dispatch (libglib-2.0.so.0)
                                                       #13 0x00007f6787e3ec88 n/a (libglib-2.0.so.0)
                                                       #14 0x00007f6787e3efa2 g_main_loop_run (libglib-2.0.so.0)
                                                       #15 0x00007f67898c13a7 gtk_main (libgtk-x11-2.0.so.0)
                                                       #16 0x0000000000432b84 n/a (/home/imalison/.cache/taffybar/taffybar-linux-x86_64)
```

**Comment by teleshoes** [comment]
@IvanMalison 
cool, it makes total sense for it to be 105, and its only WSHUD exposing it by doing something with the destroyed window, where WSS didnt do anything with it. (although it seemed like it happened when OPENING windows)

as for your proposed workaround, i already have `sortIcons=false` in my hud config, but i dont see any io action for custom sort...im sorry, i think i dont quite understand your suggestion. could you be more specific?

**Comment by IvanMalison** [comment]
@teleshoes oops I guess I assumed that travitch had merged #227, but it appears that it hasn't made it yet.

Unfortunately, until that pr is merged, there is no way to prevent the call to getGeometry

**Comment by travitch** [comment]
Sorry, didn't even see that one go by

**Comment by IvanMalison** [comment]
@teleshoes are you able to verify that setting #227 Fixes this issue when `iconSort = return . id`?

**Comment by IvanMalison** [comment]
@teleshoes I just got the issue you described, but interestingly, it is actually just one of the bars that is experiencing the issue.

When I toggle the bar using the toggle bar on monitor dbus function, the bar is fine again.

I don't think this has anything to do with getGeometry though :(

**Comment by IvanMalison** [comment]
@teleshoes Let me know if you can get on #229 and poke at this some more to see if it still happens for you.

**Comment by IvanMalison** [comment]
I just got this even with #229 applied, so it seems that that pr does not ameliorate the hang issue. The thing thats really weird about this is that its not actually crashing the process. In fact, if you are using multi-head, with taffybar on multiple monitors, the other monitor is usually fine. There is also no output at all, so I'm kind of at at a loss at how to even approach the issue, especially since there is no known repro :(

**Comment by IvanMalison** [comment]
@travitch Can you edit the name of this issue to something like:

Taffybar GTK Window Stops Refreshing in WSHUD

This issue is driving me crazy. I feel like I'm playing wack a mole with bugs in taffybar. All I want is a nice, stable experience... I thought that after #105 we would get that, but this is happening pretty regularly for me now.

**Comment by teleshoes** [comment]
@IvanMalison i havent seen this specific issue since the big x11 cleanup you did (single thread, issue 105, etc)

i am, however, still seeing crashes routinely (and only with WSHUD), so i opened a new issue for that

i REALLY hate to say this, but i instantly get a stable taffybar when i switch back to WSS
:'(

im going to try and disprove that by switching to WSS for at least a week, starting today. (ive been switching back and forth about a day at a time, and thats enough time for WSHUD to crash, but not enough to confirm that WSS doesnt crash)

if taffybar-with-WSS crashes, i will rejoice and tell you at once.

**Comment by IvanMalison** [comment]
> @IvanMalison i havent seen this specific issue since the big x11 cleanup you did (single thread, issue 105, etc)

@teleshoes That's pretty weird because I DO actually get this issue quite a bit these days :(

I've been mainly getting the crashes in #231, which seems like a memory management issue. I feel like any memory issue must be caused by IconImages.hs, but that SHOULD theoretically also affect WSS.



> i REALLY hate to say this, but i instantly get a stable taffybar when i switch back to WSS
> :'(

So your crashes are VERY frequent? As in > 1 per hour?

> if taffybar-with-WSS crashes, i will rejoice and tell you at once.

I'm not sure that would be a reason to rejoice. If WSS doesn't crash, then at least we know where the problem is. I have looked over the WSHUD code so many times and I just don't see where any memory management issue could possibly be coming from.

**Comment by teleshoes** [comment]
@IvanMalison 
>So your crashes are VERY frequent? As in > 1 per hour?

nono, i only crash about once per day in WSHUD. i was crashing 5 or 6 times a day before the last set of fixes, though, so im pretty sure 105 was the worst culprit.

its just that i literally NEVER crash on WSS, with the identical feature set being used (very different impls of course, just the same from the user's perspective)

**Comment by IvanMalison** [comment]
@teleshoes Have you tried WSS with the changes from #229 ?

I think that I may have introduced a memory management bug with the icon code changes that I made there, which would explain all of these new crashes.

I did not get the crashes from #231 (or if I did it was very infrequent because I was crashing so much due to #105).

Anyway, run on WSS with #229 for a while, and let me know if you get anything that looks like #231 

if you do, we know the issue is in IconImages, and if you don't then I guess we can assume that its in WSHUD

**Comment by teleshoes** [comment]
ok. im currently running e314808, with my two PRs (208 + 209) cherry-picked on top

**Comment by IvanMalison** [comment]
Here is a stack overflow question I asked about this issue: https://stackoverflow.com/questions/46962522/one-window-of-several-suddenly-stops-repainting-but-still-responds-to-mouse-i

**Comment by teleshoes** [comment]
yep, this was specifically about the ui-painting-thread-hanging, and thats gone.

of course the crashes still happen for me, and i believe its still true that they only happen in wshud and not wss, but i havent checked back to wss in quite awhile.

**Comment by IvanMalison** [comment]
This has started happening for me again, although its not clear to me that it has to do with WSHUD. Interestingly, the clock continues to update, but nothing else does.

@teleshoes Do you recall if that is consistent with what you experienced?

**Comment by teleshoes** [comment]
no, the clock stopped for me, too

**Comment by IvanMalison** [comment]
This definitely still happens, but its super rare. If anyone gets this frequently or predictable, please let me know

**Comment by IvanMalison** [comment]
@teleshoes hmmm I think I might have found the cause of this, even though I don't fully understand yet.

See #337
And the comment here:
https://github.com/taffybar/taffybar/blob/5ac1717213a91902166c385bb28711f9e17279f5/src/System/Taffybar/Widget/Workspaces.hs#L736

https://github.com/taffybar/taffybar/blob/5ac1717213a91902166c385bb28711f9e17279f5/src/System/Taffybar/IconImages.hs#L37

You can reproduce the issue by setting

windowIconSize to the exact size provided by some ewmh icon

**Comment by teleshoes** [comment]
oh snap! that is bizarre, bizarre, bizarre

**Comment by teleshoes** [comment]
very nice! instantly reproduced with chromium (`128x128`).
with `windowIconSize=128`, as soon as i select chromium, taffybar freezes.

this very likely explains why i havent seen the freeze in awhile: i switched `24x24` => `28x28` when i got a 1440p screen, and who uses 28x28?

**Comment by IvanMalison** [comment]
@teleshoes Well it is fixed on master. I'll probably do a release soon as this is a very serious issue.

The behavior is the same as what you saw before right? So I think we can definitively say that we found the cause, right?

**Comment by teleshoes** [comment]
looks like it to me! :D

**Comment by IvanMalison** [comment]
This still seems to happen for me. The previously found cause seems to be a red herring.

**Comment by IvanMalison** [comment]
This has gotten worse since the new method of looking up icons has been introduced. No idea why though.

**Comment by IvanMalison** [comment]
This needs to be fixed. Perhaps we should move back to the old method of icon loading.

**Comment by teleshoes** [comment]
weird, this doesnt effect me at all anymore. maybe its the by-class or by-desktop, since im not using those

**Comment by IvanMalison** [comment]
> weird, this doesnt effect me at all anymore. maybe its the by-class or by-desktop, since im not using those

Yeah, I'm pretty sure that this is what it is. I'm starting to suspect that the haskell-gi introspection might have an undocumented null somewhere, or that it is not properly taking ownership of the pixbuf it returns when it does icon lookup.

**Comment by IvanMalison** [comment]
I haven't seen this since c235869, but I forgot that I had put in a fix since #231 had gotten so much worse.

I think that c235869 did actually fix this, but it made #231 much more obvious because of the introduction of warnNotOnUI thread which happened to call threadSelf which was the main cause of that issue.
